### PR TITLE
Add GDScript `.editorconfig` rules

### PR DESCRIPTION
**Pull request by adamscott**
- Uniformize `.gd` unit test files:
  - indentation to tabs (where needed) and;
  - new lines.

### CODE REVIEW

File: modules/gdscript/tests/scripts/analyzer/errors/function_used_as_property.gd [hunk 1]
```diff
@@ -1,6 +1,5 @@
 func function():
 	pass
 
-
 func test():
```
Comment (aaronfranke) [inline modules/gdscript/tests/scripts/analyzer/errors/function_used_as_property.gd:4]: This change goes contrary to the official GDScript style guide.

Comment (YeldhamDev) [reply modules/gdscript/tests/scripts/analyzer/errors/function_used_as_property.gd:4]: There's a lot of other line removals that break the style guide as well.

Comment (adamscott) [reply modules/gdscript/tests/scripts/analyzer/errors/function_used_as_property.gd:4]: I'll remove any changes about line removals.

**Comment by adamscott** [comment]
I removed all new line changes from the PR, only to keep space indent to tabs changes.

**Review by akien-mga — APPROVED** [review]
Looks good to me. Might need a rebase to make sure new tests match the requirement?

File: modules/gdscript/tests/scripts/parser/errors/function_conflicts_constant.gd [hunk 2]
```diff
@@ -1,5 +1,4 @@
 const test = 25
 
-
```
Comment (aaronfranke) [inline modules/gdscript/tests/scripts/parser/errors/function_conflicts_constant.gd:3]: There is still this line that was removed.

**Review by aaronfranke — APPROVED** [review]

File: modules/gdscript/tests/scripts/parser/errors/function_conflicts_constant.gd (hunk 2 above)
Comment (adamscott) [reply modules/gdscript/tests/scripts/parser/errors/function_conflicts_constant.gd]: Oops! Fixed it.

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**